web/views.py

```diff
@@ -19,6 +21,27 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class TopListView(TemplateView):
+    template_name = "web/pages/top_list.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        info = (
+            Track.objects.filter(user=self.request.user)
+            .prefetch_related("artists")
+            .prefetch_related("genres")
+        )
+        context["top_artists"] = (
```

Comment:
Нужно обернуть в список, чтобы запрос происходил тут.
Иначе в шаблон передаётся QuerySet, который используется в шаблоне два раза, из-за чего происходит два запроса (при этом порядок элементов может измениться)
```suggestion
        context["top_artists"] = list(
```